web/src/main/java/org/cbioportal/web/util/StudyViewFilterApplier.java

```diff
@@ -294,27 +294,41 @@ public SampleIdentifier apply(Sample sample) {
             Map<String, List<MolecularProfile>> mapByStudyId = filteredMolecularProfiles.stream()
                     .collect(Collectors.groupingBy(MolecularProfile::getCancerStudyIdentifier));
 
-            for (List<SingleGeneQuery> geneQueries : genefilter.getSingleGeneQueries()) {
+            for (List<GeneFilterQuery> geneQueries : genefilter.getGeneQueries()) {
                 List<String> studyIds = new ArrayList<>();
                 List<String> sampleIds = new ArrayList<>();
 
-                List<String> hugoGeneSymbols = geneQueries.stream().map(SingleGeneQuery::getHugoGeneSymbol)
-                        .collect(Collectors.toList());
+                List<String> hugoGeneSymbols = geneQueries.stream()
+                    .map(org.cbioportal.model.GeneFilterQuery::getHugoGeneSymbol).collect(Collectors.toList());
+
+                Map<String, Integer> symbolToEntrezGeneId = geneService
+                    .fetchGenes(new ArrayList<>(hugoGeneSymbols),
+                        GeneIdType.HUGO_GENE_SYMBOL.name(), Projection.SUMMARY.name())
+                    .stream().collect(Collectors.toMap(x -> x.getHugoGeneSymbol(), x -> x.getEntrezGeneId()));
 
-                List<Integer> entrezGeneIds = geneService
-                        .fetchGenes(hugoGeneSymbols, GeneIdType.HUGO_GENE_SYMBOL.name(), Projection.SUMMARY.name())
-                        .stream().map(gene -> gene.getEntrezGeneId()).collect(Collectors.toList());
+                geneQueries.removeIf(
+                    q -> !symbolToEntrezGeneId.containsKey(q.getHugoGeneSymbol())
+                );
 
+                geneQueries.stream().forEach(
+                    q -> q.setEntrezGeneId(symbolToEntrezGeneId.get(q.getHugoGeneSymbol()))
+                );
```

Comment:
Well, I do not quite remember why I added this. It appears that this new logic is there to handle the event where the _geneService_ returns EntrezGeneIds for less genes (because of incomplete mapping) for the number of GeneQueries that are passed in. Normally the EntrezGeneIds were passed to the _getMutationsInMultipleMolecularProfiles_ method, effectively reducing the search space to less genes. In my PR it is the list of GeneQueries that is passed to _getMutationsInMultipleMolecularProfiles_. So, I first filter out the queries that do not relate to correctly mapped genes

Does that make sense?.

Reply:
Makes sense, thanks for explaining.